src/Component/Symfony/Routing/Loader/ResourceLoader.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Symfony\Routing\Loader;
+
+use Gedmo\Sluggable\Util\Urlizer;
+use Sylius\Component\Resource\Metadata\MetadataInterface;
+use Sylius\Component\Resource\Metadata\RegistryInterface;
+use Sylius\Component\Resource\Symfony\Bundle\Routing\Configuration;
+use Sylius\Component\Resource\Symfony\Routing\Factory\RouteFactoryInterface;
+use Symfony\Component\Config\Definition\Processor;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\Config\Loader\LoaderResolverInterface;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+use Symfony\Component\Yaml\Yaml;
+
+final class ResourceLoader implements LoaderInterface
+{
+    private RegistryInterface $resourceRegistry;
+
+    private RouteFactoryInterface $routeFactory;
+
+    public function __construct(RegistryInterface $resourceRegistry, RouteFactoryInterface $routeFactory)
```

Comment:
Would be cool to use property promotion.

```suggestion
    public function __construct(private RegistryInterface $resourceRegistry, private RouteFactoryInterface $routeFactory)
```

Reply:
yep, in another PR :dancers:

Reply:
Extracted to https://github.com/Sylius/SyliusResourceBundle/issues/752